gay.nzp.nzportable.json

```diff
@@ -0,0 +1,78 @@
+{
+    "app-id": "gay.nzp.nzportable",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "24.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "nzportable",
+    "finish-args": [
+        "--device=dri",
+        "--share=ipc",
+        "--socket=pulseaudio",
+        "--socket=x11",
+        "--env=SDL_VIDEODRIVER=x11",
+        "--env=SDL_AUDIODRIVER=pulseaudio",
+        "--share=network",
+        "--device=input",
+        "--require-version=1.16.0"
+    ],
+    "modules": [
+        {
+            "name": "nzportable-linux64",
+            "only-arches": [
+                "x86_64"
+            ],
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir -p /app/bin",
+                "mkdir -p /app/share/nzportable",
+                "# install the game binary",
+                "install -Dm755 nzportable64-sdl /app/share/nzportable/nzportable.bin",
+                "# install default.fmf",
+                "cp default.fmf /app/share/nzportable/",
+                "# install game data",
+                "mkdir -p /app/share/nzportable/nzp",
+                "for f in *; do if [ \"$f\" != \"nzportable64-sdl\" ] && [ \"$f\" != \"default.fmf\" ]; then cp -r \"$f\" /app/share/nzportable/nzp/; fi; done",
+                "echo \"nightly\" > /app/share/nzportable/nzp/version.txt",
+                "printf '%s\\n' '#!/bin/bash' '' 'DATA_DIR=\"$HOME/.var/app/gay.nzp.nzportable/data\"' 'GAME_DIR=\"$DATA_DIR/nzportable\"' 'SOURCE_DIR=\"/app/share/nzportable\"' '' 'mkdir -p \"$DATA_DIR\"' '' 'if [ ! -d \"$GAME_DIR\" ] || [ ! -f \"$GAME_DIR/nzportable.bin\" ]; then' '    echo \"Setting up NZPortable data directory...\"' '    mkdir -p \"$GAME_DIR\"' '    cp -r \"$SOURCE_DIR\"/* \"$GAME_DIR/\"' '    chmod +x \"$GAME_DIR/nzportable.bin\"' '    echo \"Setup complete.\"' 'fi' '' 'if [ \"$SOURCE_DIR/nzp/version.txt\" -nt \"$GAME_DIR/nzp/version.txt\" ]; then' '    echo \"Updating game data...\"' '    # Preserve user settings' '    if [ -f \"$GAME_DIR/nzp/user_settings.cfg\" ]; then' '        cp \"$GAME_DIR/nzp/user_settings.cfg\" /tmp/nzp_user_settings_backup.cfg' '    fi' '    # Update game files' '    cp -r \"$SOURCE_DIR/nzp\"/* \"$GAME_DIR/nzp/\"' '    # Restore user settings' '    if [ -f \"/tmp/nzp_user_settings_backup.cfg\" ]; then' '        cp /tmp/nzp_user_settings_backup.cfg \"$GAME_DIR/nzp/user_settings.cfg\"' '        rm /tmp/nzp_user_settings_backup.cfg' '    fi' '    echo \"Update complete.\"' 'fi' '' 'cd \"$GAME_DIR\"' 'exec ./nzportable.bin \"$@\"' > nzportable-launcher",
```

Comment:
NEVER EVER use a shell command to generate a shell script like that.
There is a `script` source type. Or you can include the file directly if it can't come from upstream.

This is not an obfuscation context.